src/services/completions.ts

```diff
@@ -1683,6 +1686,15 @@ function createCompletionEntry(
         hasAction = true;
     }
 
+    if (completionKind === CompletionKind.ObjectPropertyDeclaration && contextToken &&
+        findPrecedingToken(contextToken.pos, sourceFile, contextToken)?.kind !== SyntaxKind.CommaToken &&
+        (isMethodDeclaration(contextToken.parent.parent) || isSpreadAssignment(contextToken.parent) || findAncestor(contextToken.parent, (node: Node) => isPropertyAssignment(node))?.getLastToken() === contextToken ||

```

Comment:
I think the spread assignment case should be handled the same as the property assignment case, because the spread assignment can also contain any expression, like this:

```ts
const v: I = {
    ...a.b.c.d
}
```

so I think the condition here should be something like:
`findAncestor(contextToken.parent, node => isPropertyAssignment(node) || isSpreadAssignment(node))?.getLastToken()`